src/modules/Elsa.Studio.Workflows/Components/WorkflowDefinitionList/WorkflowDefinitionList.razor.cs

```diff
@@ -78,7 +83,8 @@ private async Task<TableData<WorkflowDefinitionRow>> ServerReload(TableState sta
                         publishedVersionNumber,
                         definition.Name,
                         definition.Description,
-                        definition.IsPublished);
+                        definition.IsPublished,
+                        (definition?.Links.Count(l=> l.Rel == "publish") ?? 0) == 0);
```

Comment:
I suspect it doesn't matter wether you use "publish" or "bulk-publish" to determine whether a workflow is readonly, but seeing two different bits of logic might be a bit confusing. Perhaps it would be good to introduce a small private method or even a static extension method called e.g. `IsReadOnly()` that encapsulates this logic in a single place.

Reply:
Yes, good point on the bulk vs individual - I missed that important detail 👍🏻